### PR TITLE
JavaScript DOM Cheat Sheet Added

### PR DESCRIPTION
**Pull request by TanaLandis**
**What does your Instant Answer do?**
Provides a cheat sheet for common JavaScript methods that interact with the DOM.

**What problem does your Instant Answer solve (Why is it better than organic links)?**
This is a second cheat sheet for JavaScript. The first was Core JavaScript #1558 . I am also creating cheat sheets for date methods, operators and expressions, and graphics. By creating separate cheat sheets for each, the IA provides a more specific search result and less info for the user to scan through.

**What is the data source for your Instant Answer? (Provide a link if possible)**
Over API 
http://overapi.com/javascript/
I cross referenced MDN as some of the methods on Over API have been depreciated and there were a few methods to add in.
https://developer.mozilla.org/en-US/docs/Web/JavaScript

**Why did you choose this data source?**
It was given as a source when I claimed the IA.

**Are there any other alternative (better) data sources?**
MDN (Mozilla Developer Network) is up to date and comprehensive, but contains too much info for me to consider.

**What are some example queries that trigger this Instant Answer?**
js, java script, DOM, events, methods, 

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**
developers

**Is this Instant Answer connected to a DuckDuckHack [Instant Answer idea](https://duck.co/ideas)?**
Yes - https://duck.co/ia/view/javascript_dom_cheat_sheet

**Which existing Instant Answers will this one supercede/overlap with?**
None I am aware of.

**Are you having any problems? Do you need our help with anything?**
On the duck.co side, how do I link all the cheat sheets to one IA? It was supposed to be one cheat sheet but there is so much info there has to be multiple.

**Where did you hear about DuckDuckHack? (For first time contributors)**
GDI Lehigh Valley

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**
javascript dom

## Checklist
Please place an 'X' where appropriate.

```
- [X] Added metadata and attribution information (https://duck.co/duckduckhack/metadata)
- [] Wrote test file and added to t/ directory (https://duck.co/duckduckhack/goodie_tests)
- [] Verified that instant answer adheres to design guidelines (https://duck.co/duckduckhack/design_styleguide)
- [] Verified that instant answer adheres to code styleguide (https://duck.co/duckduckhack/code_styleguide)
- [] Tested cross-browser compatibility

    Please let us know which browsers/devices you've tested on:
    - Windows 8
        - [X] Google Chrome
        - [] Firefox
        - [] Opera
        - [] IE 10

    - Windows 7
        - [] Google Chrome
        - [] Firefox
        - [] Opera
        - [] IE 8
        - [] IE 9
        - [] IE 10

    - Windows XP
        - [] IE 7
        - [] IE 8

    - Mac OSX
        - [] Google Chrome
        - [] Firefox
        - [] Opera
        - [] Safari

    - iOS (iPhone)
        - [] Safari Mobile
        - [] Google Chrome
        - [] Opera

    - iOS (iPad)
        - [] Safari Mobile
        - [] Google Chrome
        - [] Opera

    - Android
        - [] Firefox
        - [] Native Browser
        - [] Google Chrome
        - [] Opera
```

-----
IA Page: https://duck.co/ia/view/javascript_dom_cheat_sheet

### CODE REVIEW

File: share/goodie/cheat_sheets/json/javascript-dom.json [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+    "name": "JavaScript and the DOM",
+    "description": "Explanation of JavaScript methods which interact with the DOM",
+    "metadata": {
+        "sourceName": "Over API",
+        "sourceUrl": "http://overapi.com/javascript/"
+    },
+    "aliases": [
+        "js", "java script", "methods", "events"  
+    ],
+    "template_type": "code",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/javascript-dom.json]: I think this one will be more legible if we switch to the `reference` template:

screen shot 2015-09-25 at 18 23 27

File: share/goodie/cheat_sheets/json/javascript-dom.json [hunk 2]
```diff
@@ -0,0 +1,238 @@
+{
+    "id": "javascript_dom_cheat_sheet",
+    "name": "JavaScript and the DOM",
+    "description": "Explanation of JavaScript methods which interact with the DOM",
+    "metadata": {
+        "sourceName": "Over API",
+        "sourceUrl": "http://overapi.com/javascript/"
+    },
+    "aliases": [
+        "js", "java script", "methods", "events"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/javascript-dom.json]: Seeing as the previous cheat sheet you made uses the "js" and "javascript" aliases, this one will have to use others. For now, the triggers and aliases need to be unique across Cheat Sheets as we currently only show 1 cheat sheet for any given query.

We're working on showing more Instant Answers and this "series" of JS Cheat Sheets might prove as a perfect use-case. We have considered showing a tile view with several cheat sheets that match, but for now we're just showing 1.

Let's use aliases like `"js dom", "js methods", "js events", "javascript events", "javascript methods"`

File: share/goodie/cheat_sheets/json/javascript-dom.json (hunk 1 above)
Comment (TanaLandis) [inline share/goodie/cheat_sheets/json/javascript-dom.json]: I agree!

**Comment by moollaza** [comment]
@TanaLandis thanks a lot for contributing, we really appreciate it! I've left you some feedback that needs to be implemented. Once things are updated please let us know by leaving a comment, so we can review this again. If you have any questions, feel free to ask them here!

File: share/goodie/cheat_sheets/json/javascript-dom.json (hunk 2 above)
Comment (TanaLandis) [inline share/goodie/cheat_sheets/json/javascript-dom.json]: Got it. I assumed multiple IAs would show. In my mind it was the perfect reference as multiple sheets. This changes things a bit. I'm going to think it through a bit more before making any revisions. Glad you could see what I was trying to do :)!

**Comment by TanaLandis** [comment]
@moollaza I used terminal template to keep more of a visual consistency between this cheat sheet and JavaScript Core. Your thoughts on the aliases between JavaScript Core and JavaScript DOM cheat sheets? I placed methods as an alias with JavaScript Core because its in the name and description. Also thought it would be more likely for someone to query "javascript methods" then "javascript dom methods". 
Travis CI build is failing. Is this a problem?

**Comment by moollaza** [comment]
@TanaLandis apologies for the delay! I'll be taking a look at this today!

**Comment by moollaza** [comment]
> Travis CI build is failing. Is this a problem?

@TanaLandis I checked your JSON and it looks correct so the fault is not yours (I think) but if you `git pull` the most recent changes to your branch, you should see the tests pass again.

File: share/goodie/cheat_sheets/json/javascript.json [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
     "name": "JavaScript Methods",
-    "description": "Explanation of core JavaScript methods",
+    "description": "Explanation of core JavaScript methods. For JavaScript DOM methods, query \"javascript DOM cheat sheet\".",
     "metadata": {
         "sourceName": "OverAPI",
         "sourceUrl": "http://overapi.com/javascript/"
     },
     "aliases": [
-        "js", "java script" 
+        "js", "java script", "javascript methods", "js methods", "java script methods",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/javascript.json]: @TanaLandis I found the problem! The trailing comma (`,`) at the end of this line isn't valid JSON syntax and so it's causing the test to fail. Please fix that and then we'll be able to deploy this to Beta

**Comment by moollaza** [comment]
@TanaLandis Ping! I'm just checking in to see if you're still interested in finishing this one up? We'd love to see this go live!

There's absolutely no rush, just wanted to make sure you're not stuck on anything.

---

BTW, thanks for doing such a great job so far! :cake: We really appreciate you taking the time to contribute. As a thanks, after this is merged we'd like to write a short blog post about you and your Instant Answer.

If you're interested please send an email to open@duckduckgo.com, with the subject "DuckDuckHack Dev Appreciation".

Please provide your name and include a link to this pull request as well.

We'll get in touch for more details!

**Comment by TanaLandis** [comment]
@moollaza Thank you for the reminder! My apologies it is still hanging there, especially with such a simple fix. I want to and will wrap it up by the end of next week.

**Comment by TanaLandis** [comment]
@moollaza There are 2 cheat sheets in this one pull request - javascript.json and javascript-dom.json . Can you instruct me how to resolve this if it is an issue? I sort of recall that it was, but don't remember what to do.

**Comment by moollaza** [comment]
@TanaLandis I've continued this for you in #2099 -- I didn't carry along your updates to Javascript.json though -- would you like to make a new Pull Request to update the aliases for that again?

Apologies about the delays and confusion with getting this merged!

**Pull request closed without merging**